src/main/java/formflow/library/FileController.java

```diff
@@ -276,34 +267,44 @@ RedirectView delete(
   }
 
   /**
-   * @param httpSession  The current HTTP session
    * @param submissionId The submissionId of the file to be downloaded
    * @param fileId       The UUID of the file to be downloaded.
+   * @param flow         The name of the current (active) flow
+   * @param httpSession  The current HTTP session
+   * @param request      The HttpServletRequest
    * @return ON SUCCESS: ResponseEntity with a response body that includes the file.
    * <p>ON FAILURE: A ResponseEntity returns an HTTP error code</p>
    */
-  @GetMapping("/file-download/{submissionId}/{fileId}")
+  @GetMapping("/file-download/{flow}/{submissionId}/{fileId}")
   public ResponseEntity<StreamingResponseBody> downloadSingleFile(
-      HttpSession httpSession,
       @PathVariable String submissionId,
       @PathVariable String fileId,
+      @PathVariable String flow,
+      HttpSession httpSession,
       HttpServletRequest request
   ) {
     log.info("GET downloadSingleFile (url: {}): submissionId: {} fileId {}", request.getRequestURI().toLowerCase(), submissionId,
         fileId);
-    if (!submissionId.equals(httpSession.getAttribute("id").toString())) {
+
+    if (!submissionId.equals(getSubmissionIdForFlow(httpSession, flow).toString())) {
+      log.error("Submission ID handed in doesn't match the one associated with the file.");
```

Comment:
Maybe include the ID handed in with the log?

Maybe something like "There was an attempt to download a file with submission ID {ID} that does not match the submission of the file being downloaded."

Reply:
I wasn't sure how much information we wanted to expose here.  What was your philosophy on this when you created it?  I was thinking it might be handy to have the fileId in this msg as well.

Reply:
My thoughts were around providing the most information as to be helpful to SNEs if they are trying to debug a problem, or better understand what happened that lead to a file not being downloaded. In theory they should be the only ones seeing these logs.

In terms of exposing more or less, are you thinking about an outside actor gaining access to the logs?

Reply:
Yeah, an outside actor scenario.  I thought at one point folks were hesitant to include more file information in the logs, though I don't see it as a bad thing for us to do. 

I can include more information here, like the file id, so someone reading the log would have more to debug with.

Reply:
There may be a question there for Dimitri. My thought was we didn't want to expose any PII in the logs (which things like original file names might include). Though I suppose if someone gained access to the logs they could see submission IDs and session IDs and potentially put them together to download information, though I think if someone gained log access we might have bigger problems?